eng/tools/dependency-testing/index.js

```diff
@@ -0,0 +1,236 @@
+let argv = require("yargs")
+  .options({
+    "artifact-name": {
+      type: "string",
+      describe:
+        "name of the artifact to be incremented (e.g. azure-keyvault-secrets), will be translated to @azure/(package) format",
+      demandOption: true
+    },
+    "repo-root": {
+      type: "string",
+      default: "../../../",
+      describe: "root of the repository (e.g. ../../../)",
+      demandOption: true
+    },
+    "min-max": {
+      type: "string",
+      describe: "whether you want to test max or min version of dependencies",
+      demandOption: true
+    },
+    "dry-run": {
+      type: "boolean"
+    }
+  })
+  .help().argv;
+
+const fs = require("fs");
+const path = require("path");
+const semver = require("semver");
+const packageUtils = require("eng-package-utils");
+//var spawnSync = require("child_process").spawnSync, child;
+var crossSpawn = require("cross-spawn");
+
+function outputTestPath(targetPackagePath) {
+  const testPath = path.join(targetPackagePath, "test");
+  console.log(`##vso[task.setvariable variable=PackageTestPath]${testPath}`)
+  console.log(`Emitted variable "PackageTestPath" with content: ${testPath}`)
+}
+
+async function insertPackageJson(packageJsonContents, targetPackagePath, targetPackageName, minMax) {
+  //console.log("package path = " + targetPackagePath);
+  //console.log(packageJsonContents["devDependencies"]);
+  const testPath = path.join(targetPackagePath, "test");
+  var templateJson = await packageUtils.readFileJson("./templates/package.json");
+  console.log(templateJson);
+  var testPackageJson = templateJson;
+  testPackageJson.name = packageJsonContents.name.replace("@azure/", "azure-") + "-test";
+  testPackageJson.devDependencies = packageJsonContents.devDependencies;
+
+  var versionTags = await getVersionsWithTag(targetPackageName);
+  depList = {};
+  if (versionTags) {
+    console.log(versionTags);
+    var lines = versionTags.split("\n");
```

Comment:
Put a comment in here that mentions `npm dist-tag` doesn't output in JSON so we are forced to parse.